docs/elementExtensions.md

```diff
@@ -0,0 +1,73 @@
+# Element extensions developer guide
+
+Extensions are a way to add custom logic and extra complexity to already existing elements.  Each element has the ability to load relevant extensions and as such, extensions should be custom tailored to one specific element.
```

Comment:
I think this is what I thought PR https://github.com/PrairieLearn/PrairieLearn/pull/2384#discussion_r411444013 was?

Reply:
Ah, let me clarify a little bit.  Course elements allow someone to replace a whole element with a local copy in their course, which allows an easy way to make changes but then you end up with duplicate copies. 
What I'm proposing here is allowing elements to almost create their own developer interfaces that courses can make extensions for, I was initially thinking this could be used for the `pl-drawing` element if people wanted to make custom drawing objects, though this can be used by any large elements in the future as needed.

Reply:
@nicknytko excellent. We're on the same page then! 

> but existing elements will need to add support for extensions to be able to use them.

In this case, I think prior to merging the PR, there should be an extension interface applied to at least one base element, `pl-drawing` would suffice in this case.

Reply:
Agreed, currently working on that for `pl-drawing` in a different branch but I think that may involve quite a bit of refactoring :slightly_smiling_face: 

How about the `pl-python-variable` element?  I think one could argue for the use case of adding new renderable types to that element.

Reply:
Discussed with @mfsilva22, added an example extension to the `pl-graph` elem